src/ast/rewriter/bv_rewriter.cpp

```diff
@@ -2921,6 +2925,21 @@ br_status bv_rewriter::mk_distinct(unsigned num_args, expr * const * args, expr_
     return BR_DONE;     
 }
 
+br_status bv_rewriter::mk_bvsmul_overflow(unsigned num, expr * const * args, expr_ref & result) {
+    SASSERT(num == 2);
+    result = m.mk_or(
+            m.mk_not(m_util.mk_bvsmul_no_ovfl(args[0], args[1])),
+            m.mk_not(m_util.mk_bvsmul_no_udfl(args[0], args[1]))
+    );
+    return BR_REWRITE1;
+}
+
+br_status bv_rewriter::mk_bvumul_overflow(unsigned num, expr * const * args, expr_ref & result) {
+    SASSERT(num == 2);
+    result = m.mk_not(m_util.mk_bvumul_no_ovfl(args[0], args[1]));
+    return BR_REWRITE1;
```

Comment:
I made a mistake, the result should be BR_REWRITE2.
It instructs to not just the outer negation, but the arguments to it.

Reply:
Ok!